numpy/core/src/umath/scalarmath.c.src

```diff
@@ -167,7 +167,7 @@ static NPY_INLINE int
     }
 #if @neg@
     else if (b == -1 && a == NPY_MIN_@NAME@) {
-        *out = a / b;
+        *out = NPY_MIN_@NAME@;
```

Comment:
I think https://github.com/numpy/numpy/pull/21507 will cover it so we can remove this here perhaps? All though I'm happy to rebase if it goes in first.